File: api/src/Domain/Settings/DataTransferObjects/ServiceUpdate.cs
using System.Collections.Generic;

namespace DetailingArsenal.Domain.Settings {
    public class ServiceUpdate : IDataTransferObject {
        public string Name { get; }
        public string? Description { get; }
        public ServicePricingMethod PricingMethod { get; }
        public List<ServiceConfigurationUpdate> Configurations { get; }

        public ServiceUpdate(string name, string? description, ServicePricingMethod pricingMethod, List<ServiceConfigurationUpdate> configurations) {
            Name = name;
            Description = description;
            PricingMethod = pricingMethod;
            Configurations = configurations;
        }
    }
}

File: app/src/api/settings/data-transfer-objects/service-update.ts
import { ServicePricingMethod } from '@/api/settings/data-transfer-objects/service-pricing-method';

export type ServiceUpdate = {
    id: string;
    name: string;
    description?: string;
    pricingMethod: ServicePricingMethod;
    configurations: { vehicleCategoryId: string | null; price: number; duration: number }[];
};
